pkg/utils/secrets/basic_auth_hash.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
```

Comment:
Isn't it possible include this functionality into the existing `BasicAuth` struct? That's a lot of duplicated code now.

Reply:
It is possible, but with this modification, all secrets will have an extra field(`hashedPassword`), and most will not need it. 
If you want i can do it :)

Reply:
Can't you make it configurable? E.g., for the control plane secrets, not everything needs a kubeconfig. This is configurable there as well.

Reply:
Can you review if that was what you meant